### PR TITLE
Update spell.yml

### LINKED ISSUE #31602: [Tooling] update pyspelling in SDK to support camel-case
### Reproduction steps / Feature

newer versions fixed the camel-case option to aspell. The spec repo uses this version. We should update the SDK to do the same. Then we can remove some of the excluded files from the spell checker.

### Platform

python

### Platform Version(s)

_No response_

### Type

CI tested

### (Optional) If manually tested please explain why this is only manually tested

_No response_

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by kliao-csa**
Update the spell checker used in pyspelling checker to rojopolis/spellcheck-github-actions@0.36.0 that is used in chip-test-plans

### CODE REVIEW

**Comment by andy31415** [comment]
@kliao-csa - this should have been going through a regular review process as it is not a separate example/tool.

File: .github/workflows/spell.yml [hunk 1]
```diff
@@ -25,19 +25,20 @@ on:
...
-        name: Check Spelling - reviewdog
-        runs-on: ubuntu-latest
-        steps:
-            - name: Checkout
-              uses: actions/checkout@v4
-            - uses: reviewdog/action-misspell@v1
-              with:
-                  github_token: ${{ secrets.GITHUB_TOKEN }}
+#    Seems redundant; removed as pyspelling is customized while this is not
+#    check-reviewdog:
```
Comment (andy31415) [inline .github/workflows/spell.yml:29]: This comment staying in the source code will lose context. Pyspelling is not there anymore, only spellcheck-github-actions.

File: .github/workflows/spell.yml [hunk 2]
```diff
@@ -25,19 +25,20 @@ on:
...
+#              with:
+#                  github_token: ${{ secrets.GITHUB_TOKEN }}
     check-spellcheck:
         name: Check Spelling - pyspelling
         runs-on: ubuntu-latest
         steps:
             - name: Checkout
               uses: actions/checkout@v4
-            - uses: igsekor/pyspelling-any@v1.0.4
+            - uses: rojopolis/spellcheck-github-actions@0.36.0
```
Comment (andy31415) [inline .github/workflows/spell.yml:44]: This makes the name above not valid anymore it seems. Is this still pyspelling underneath? should we update to spellcheck?

We may want some docs on what files it uses (e.g. if it still uses .spellcheck.yml)

Comment (kliao-csa) [reply .github/workflows/spell.yml:44]: This is the pyspelling distrib used in the spec repo. It uses the same files (.spellcheck.yml) in the same locations.

File: .github/workflows/spell.yml (hunk 1 above)
Comment (kliao-csa) [reply .github/workflows/spell.yml:29]: Ok, I'll remove this comment and clarify in some documentation.

**Pull request merged**